crates/solver/src/solver/http_solver/settlement.rs

```diff
@@ -32,11 +33,27 @@ pub fn convert_settlement(
     }
 }
 
+#[derive(Clone, Debug)]
+#[cfg_attr(test, derive(PartialEq))]
+enum Execution {
+    ExecutionAmm(ExecutedAmm),
+    ExecutionCustomInteraction(InteractionData),
+}
+
+impl Execution {
+    fn coordinates(&self) -> Option<ExecutionPlanCoordinatesModel> {
+        match self {
+            Execution::ExecutionAmm(executed_amm) => executed_amm.exec_plan.clone(),
+            Execution::ExecutionCustomInteraction(interaction) => interaction.exec_plan.clone(),
+        }
+    }
+}
+
 // An intermediate representation between SettledBatchAuctionModel and Settlement useful for doing
 // the error checking up front and then working with a more convenient representation.
 struct IntermediateSettlement {
     executed_limit_orders: Vec<ExecutedLimitOrder>,
-    executed_amms: Vec<ExecutedAmm>,
+    executions: Vec<Execution>,
```

Comment:
nit: maybe there should be a comment here saying that this is already sorted by execution coordinate (otherwise it would be nice to re-sort before actually appending it to the execution plan).